test_simulation.py

```diff
@@ -4,55 +4,77 @@
 import pytest
 
 from gears import Gear, GearTrain
-# codify gear example into a test
+# separate tests for readability
+# arrange, act, assert
 
-def test_geartrain_implementation_in_simnode():
-    g0 = Gear(10)
-    g1 = Gear(20)
-    g2 = Gear(10)
+@pytest.fixture
+def gear1():
+    gear = Gear(10)
+    gear.probability_of_breaking_per_tooth_click = 0 # NDH: having to move this into declaration feels like anti pattern
```

Comment:
Simnodes should probably be deterministic on declaration. If we end up with like 500,000+ nodes at runtime, updating a bunch of properties for determinism wouldnt scale too well within the application.

Reply:
Agreed. Although there are some circumstances in which an internal or external influence could alter a property of the node. E.g. wear & tear on machine parts. Setter methods maybe?